Update introduction

richlander opened the pull request:
- Reduce the size of topic.
- Focus on key value props.
- Remove content that doesn't really matter.

Some links are missing and still need to be added.

* [Current page](https://docs.microsoft.com/dotnet/core/introduction)
* [Updated page -- internal only](https://review.docs.microsoft.com/en-us/dotnet/core/introduction?branch=intro)
* [Rendered view](https://github.com/dotnet/docs/blob/intro/docs/core/introduction.md)

richlander committed:
Update introduction

jkotas commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
 recommendations: false
 ---
-# What is .NET? Introduction and overview
+# What is .NET?
```
https://www.google.com/search?q=What+is+.NET will point you to https://dotnet.microsoft.com/en-us/learn/dotnet/what-is-dotnet

How do we think about this doc vs. the content that the search engines actually point you to?

richlander replied:
Should we be writing this topic differently, then?

tdykstra replied:
By limiting the range of content in the article, the PR as it stands seems to aim at the dotnet.microsoft.com goal of helping people "decide if they want to learn the platform" whereas the more comprehensive introduction offered by the article as it stands gets them started "learning the platform," by introducing more topics and linking to documentation that goes into more depth on those concepts.

BillWagner commented:
I've added @mairaw and @BethMassi to review. I'd like them to make sure we all agree to remove the headings of "Free", "Open Source" and "Cross platform".

jkotas commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+# What is .NET?
 
-.NET is a free, open-source development platform for building many kinds of apps, such as:
+.NET is a free and open-source development platform for building any type of [app](tutorials/with-visual-studio-code.md). .NET is built on a [high-performance runtime](https://devblogs.microsoft.com/dotnet/category/performance/) that is used in production by many [high-scale apps](https://devblogs.microsoft.com/dotnet/category/app/). It is built with the community on [GitHub](https://github.com/d ...
```
"platform for building any type of app" is our north star, but I am not sure whether it is a fair statement to make.  I like more the wording used on https://dotnet.microsoft.com/en-us/learn/dotnet/what-is-dotnet .

richlander replied:
```suggestion
NET is a free, cross-platform, open source developer platform for building many different types of [applications](tutorials/with-visual-studio-code.md). .NET is built on a [high-performance runtime](https://devblogs.microsoft.com/dotnet/category/performance/) that is used in production by many [high-scale apps](https://devblogs.microsoft.com/dotnet/category/app/). It is built with the community on [GitHub](https://github.com/dotnet/core).
```

richlander committed:
Update introduction.md

gewarren commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+## Runtime Technology
 
-You can create .NET apps for many operating systems, including:
+The .NET [Common Language Runtime (CLR)](../standard/clr.md) is a low-level cross-platform runtime. The CLR provides type and memory safety, handles memory allocation and garbage collection, thread creation and management, native API interop, and many other functions.
```
```suggestion
The .NET [Common Language Runtime (CLR)](../standard/clr.md) is a low-level, cross-platform runtime. The CLR provides type and memory safety, handles memory allocation and garbage collection, creates and manages threads, and affords native API interoperability and many other functions.
```

gewarren replied:
The grammar wasn't quite right, so I tried to fix it.

gewarren commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+## Libraries and Frameworks
 
-## Advanced scenarios
+.NET includes libraries and frameworks for many topics and app types. This ranges from data structures like string and Dictionary to utility functions like HTTP and RegEx to app frameworks like Windows Forms and ASP.NET Core.
```
```suggestion
.NET includes libraries and frameworks for many topics and app types. These affordances range from data structures like `string` and `Dictionary`, to utility functions like HTTP and RegEx, to app frameworks like Windows Forms and ASP.NET Core.
```

gewarren commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+## Tools
 
-Every operating system includes an application programming interface (API) that provides system services. .NET provides several ways to call those APIs.
+The [.NET SDK](sdk.md) includes a set of [CLI](tools/index.md) tools, including [MSBuild](/visualstudio/msbuild/msbuild) build engine, the [Roslyn](https://github.com/dotnet/roslyn) (C# and VB) compiler, and the [F#](https://github.com/microsoft/visualfsharp) language compiler. Most commands are run from `dotnet`. The CLI tools can be used for local development and continuous integration.
```
```suggestion
The [.NET SDK](sdk.md) includes a set of [CLI](tools/index.md) tools, including the [MSBuild](/visualstudio/msbuild/msbuild) build engine, the [Roslyn](https://github.com/dotnet/roslyn) (C# and Visual Basic) compiler, and the [F#](https://github.com/microsoft/visualfsharp) language compiler. Most commands are run from `dotnet`. The CLI tools can be used for local development and continuous integration.
```

gewarren commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+The [.NET SDK](sdk.md) includes a set of [CLI](tools/index.md) tools, including [MSBuild](/visualstudio/msbuild/msbuild) build engine, the [Roslyn](https://github.com/dotnet/roslyn) (C# and VB) compiler, and the [F#](https://github.com/microsoft/visualfsharp) language compiler. Most commands are run from `dotnet`. The CLI tools can be used for local development and continuous integration.
 
-The main way to interoperate with native APIs is via "platform invoke" or P/Invoke for short. P/Invoke is supported across Linux and Windows platforms. A Windows-only way of interoperating is known as "COM interop," which works with [COM components](/cpp/atl/introduction-to-com) in managed code. It's built on top of the P/Invoke infrastructure, but it works in subtly different ways.
+[Visual Studio](https://visualstudio.microsoft.com/) family IDEs offer excellent support for .NET and C#, F#, and VB languages.
```
```suggestion
The [Visual Studio](https://visualstudio.microsoft.com/) family of IDEs offer excellent support for .NET and the C#, F#, and Visual Basic languages.
```

gewarren commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+[Visual Studio](https://visualstudio.microsoft.com/) family IDEs offer excellent support for .NET and C#, F#, and VB languages.
 
-For more information, see [Native interoperability](../standard/native-interop/index.md).
+[GitHub Actions](https://github.com/features/actions), [GitHub Codespaces](https://github.com/features/codespaces) and [GitHub security features](https://github.com/features/security) support .NET.
```
```suggestion
[GitHub Actions](https://github.com/features/actions), [GitHub Codespaces](https://github.com/features/codespaces), and [GitHub security features](https://github.com/features/security) support .NET.
```

gewarren reviewed (commented):
There's a lot of info being deleted - should we include that information in a separate article? Or do you feel that it's covered elsewhere already?

richlander committed:
Update docs/core/introduction.md

Co-authored-by: Jan Kotas <jkotas@microsoft.com>

richlander commented:
The deleted info mostly isn't useful. It's better to delete it all and come back to this PR for a source of info if we need to later.

richlander committed:
Apply suggestions from code review

Co-authored-by: Genevieve Warren <24882762+gewarren@users.noreply.github.com>

mairaw commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
-* [Machine learning](../machine-learning/index.yml)
-* [Console apps](tutorials/with-visual-studio-code.md)
-* [Windows services](/aspnet/core/host-and-deploy/windows-service)
+You can use .NET for client apps, for [desktop](/dotnet/desktop/), [mobile](/xamarin/), [browser](aspnet/core/blazor), and [games](https://dotnet.microsoft.com/apps/games).
```
should we say web instead browser? Also, with Blazor now coming into play with hybrid apps, should we link to aspnet/core instead?

richlander replied:
I just checked. They are called "client web" apps. We're already talking about client apps, so I guess "web" it is.

https://dotnet.microsoft.com/en-us/apps/aspnet/web-apps/blazor

```suggestion
You can use .NET for client apps, for [desktop](/dotnet/desktop/), [mobile](/xamarin/), [web](aspnet/core/blazor), and [games](https://dotnet.microsoft.com/apps/games).
```

mairaw commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+You can use .NET for [cloud apps](../architecture/cloud-native/index.md), [Web apps and APIs](/aspnet/core/introduction-to-aspnet-core#recommended-learning-path), and [serverless functions](/azure/azure-functions/functions-create-first-function-vs-code?pivots=programming-language-csharp). .NET cloud apps are competitively measured by the [techempower benchmark](https://www.techempower.com/benchma ...
 
-With .NET, your code and project files look and feel the same no matter which type of app you're building. You have access to the same runtime, API, and language capabilities with each app.
+.NET can be used for many other apps, like [IoT](dotnet/iot/), [Tools](/dotnet/core/tools/global-tools), [machine learning](../machine-learning/index.yml), and [NativeAOT](https://github.com/dotnet/runtime/issues/61231).
```
should we add NativeAOT after it launches? We are also not talking about MAUI here, although we should add it once it ships.

richlander replied:
```suggestion
.NET can be used for many other apps, like [IoT](dotnet/iot/), [Tools](/dotnet/core/tools/global-tools), and [machine learning](../machine-learning/index.yml).
```

tdykstra commented:
> The deleted info mostly isn't useful. It's better to delete it all and come back to this PR for a source of info if we need to later.
> 
> Or put a different way, none of it belongs in an intro about .NET. If the same content is missing from a more appropriate place in the docset, then that's probably already a problem.

If I were a developer new to .NET, the deleted info would help me understand what .NET has to offer and give me an overview of ways that .NET differs from other development platforms, as well as provide an overview of topics I might want to investigate in greater detail. There are a few places where details could be left to "for more information, see ..." links, but the high-level explanations are arguably useful to introduce .NET to an audience unfamiliar with it. As the article stands now, the latest user satisfaction rating is 90% on 10 responses, which is a higher "helpful" rating than all but a very few .NET docs. I think it's reasonable to attribute that high rating in part to the information in the deleted sections.

richlander committed:
Apply suggestions from code review

Co-authored-by: Maira Wenzel <mairaw@microsoft.com>

richlander commented:
That's a good point @tdykstra. We should decide what we want this topic to be. It can be any of:

- A brief introduction to the platform (what the PR represents, currently).
- A complete introduction to the platform and related high-level concepts.

There are likely some mid-points in-between.

My biggest issue is that the current document is just a mis-mash on content that clearly came from different places. It's not written with any real intent. I can see that some of it came from me, so I'm A-OK with being critical of it.

> A NuGet package is a *.zip* file with the `.nupkg` extension that contains compiled code (DLLs), other files related to that code, and a descriptive manifest that includes information like the package's version number.

That's a perfect example. That content has no place on an in introduction to a platform. Sure, some people might find it interesting, but it is three levels below what you need to know to develop a basic understanding of the platform.

I'm happy to continue working on this PR to make it cover more concepts. That said, we need to resolve the point that Bill raised first. https://github.com/dotnet/docs/pull/29275#discussion_r864801277

IEvangelist commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+# What is .NET?
 
-.NET is a free, open-source development platform for building many kinds of apps, such as:
+NET is a free, cross-platform, open source developer platform for building many different types of [applications](tutorials/with-visual-studio-code.md). .NET is built on a [high-performance runtime](https://devblogs.microsoft.com/dotnet/category/performance/) that is used in production by many [high-scale apps](https://devblogs.microsoft.com/dotnet/category/app/). It is built with the community o ...
```
```suggestion
NET is a free, cross-platform, open source developer platform for building many different types of [applications](tutorials/with-visual-studio-code.md). .NET is built on a [high-performance runtime](https://devblogs.microsoft.com/dotnet/category/performance/) that is used in production by many high-scale apps. It is built with the community on [GitHub](https://github.com/dotnet/core).
```

This link doesn't work for me:

https://devblogs.microsoft.com/dotnet/category/app/

richlander replied:
```suggestion
NET is a free, cross-platform, open source developer platform for building many different types of [applications](tutorials/with-visual-studio-code.md). .NET is built on a [high-performance runtime](https://devblogs.microsoft.com/dotnet/category/performance/) that is used in production by many [high-scale apps](https://devblogs.microsoft.com/dotnet/category/developer-stories/). It is built with the community on [GitHub](https://github.com/dotnet/core).
```

IEvangelist commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+.NET can be used for many other apps, like [IoT](dotnet/iot/), [Tools](/dotnet/core/tools/global-tools), and [machine learning](../machine-learning/index.yml).
 
-## Cross platform
+## Runtime Technology
```
```suggestion
## Runtime technology
```

IEvangelist commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+## Runtime Technology
 
-You can create .NET apps for many operating systems, including:
+The .NET [Common Language Runtime (CLR)](../standard/clr.md) is a low-level, cross-platform runtime. The CLR provides type and memory safety, handles memory allocation and garbage collection, creates and manages threads, and affords native API interoperability and many other functions.
```
```suggestion
The .NET [Common Language Runtime (CLR)](../standard/clr.md) is a low-level, cross-platform runtime. The CLR provides type and memory safety, it handles memory allocation and garbage collection, creates and manages threads, and affords native API interoperability and many other functions.
```

IEvangelist commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
-* iOS
-* tvOS
-* watchOS
+.NET apps are compiled to a compact code format that can be supported on any operating system or chip hardware. Most .NET apps use APIs that are supported in multiple environments, requiring only the .NET
+runtime to run.
```
```suggestion
.NET apps are compiled to a compact code format that can be supported on any operating system or chip hardware. Most .NET apps use APIs that are supported in multiple environments, requiring only the .NET runtime to run.
```

IEvangelist commented on docs/core/introduction.md:
```diff
@@ -6,404 +6,52 @@ ms.date: 02/24/2022
+* [Visual Basic](../visual-basic/index.yml) uses a more verbose syntax that is closer to ordinary human language. It can be an easier language to learn for people new to programming.
 
-For more information about .NET terminology, see the [.NET glossary](../standard/glossary.md).
+## Libraries and Frameworks
```
```suggestion
## Libraries and frameworks
```

richlander committed:
Apply suggestions from code review

Co-authored-by: David Pine <david.pine@microsoft.com>

richlander committed:
Add more content

richlander commented:
I added a bunch of content back and rewrote some of it. Let's see if this brings us closer. Once we have an agreement on a direction, I can work to finish this topic off.

richlander committed:
Add unsafe code to features

richlander committed:
Add binary distributions

richlander committed:
Update deployment section

richlander committed:
Change 'development' title

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +73,164 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+The [Visual Studio](https://visualstudio.microsoft.com/) family of IDEs offer excellent support for .NET and the C#, F#, and Visual Basic languages.
 
-* **.NET**
+[GitHub Codespaces](https://github.com/features/codespaces), and [GitHub security features](https://github.com/features/security) support .NET.
```
```suggestion
[GitHub Codespaces](https://github.com/features/codespaces) and [GitHub security features](https://github.com/features/security) support .NET.
```

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +73,164 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
 
-Every operating system includes an application programming interface (API) that provides system services. .NET provides several ways to call those APIs.
+* *Self-contained* apps includes the .NET runtime and dependent libraries. They can be [single-file](deploying/single-file/overview) or multi-file. Users of the application can run it on a machine that doesn't have the .NET runtime installed. Self-contained apps always target a single operating system and chip configuration.
+* *Framework-dependent* apps require a compatible version of the .NET runtime, typically installed globally. Framework-dependent apps can be published for a single operating system and chip configuration or as "portable" targeting all supported configurations.
```
```suggestion
* *Framework-dependent* apps require a compatible version of the .NET runtime, typically installed globally. Framework-dependent apps can be published for a single operating system and chip configuration or as "portable," targeting all supported configurations.
```

tdykstra reviewed (commented):
Looks good. I left some suggestions from an initial pass through the latest version that has some of the deleted content restored, and I'll take another look tomorrow. I do think the overloaded terms section contained information that would be helpful to someone new to .NET documentation and terminology.

BethMassi commented on docs/core/introduction.md:
```diff
@@ -6,169 +6,63 @@ ms.date: 02/24/2022
+
+**Cross-platform client apps**
+
 * [Mobile apps](https://dotnet.microsoft.com/learn/xamarin/hello-world-tutorial/intro)
```
```suggestion
* [Mobile apps](https://dotnet.microsoft.com/apps/maui)
```

BethMassi commented on docs/core/introduction.md:
```diff
@@ -6,169 +6,63 @@ ms.date: 02/24/2022
+**Cross-platform client apps**
+
 * [Mobile apps](https://dotnet.microsoft.com/learn/xamarin/hello-world-tutorial/intro)
-* Desktop apps
```
I'd add this back and link to https://dotnet.microsoft.com/apps/desktop which is being updated to showcase MAUI and Blazor (PR is being reviewed and should be live shortly). "Desktop" can mean web or native.

richlander replied:
This was confusing since I moved that content elsewhere. I added the link back.

https://github.com/dotnet/docs/pull/29275/commits/341d96fbf21342f0b1bbeec2c69ddf017b20a1fb

BethMassi replied:
My point is desktop development is cross platform so should have a link here. The desktop topic on the dot.net site will skew towards cross platform clients.

richlander replied:
I understand now.

https://github.com/dotnet/docs/pull/29275/commits/70c9ee2edc7c6855b7d24623b6e21bcf69f89bc6

richlander replied:
Now that I look at this more closely, I'm confused. IMO, docs should only link to the marketing website for downloads not for information. The marketing website website should link for docs for more information. The back-and-forth suggests we've got the information architecture wrong. That's right isn't it?

richlander commented:
On the "overloaded terms", I get how there is value there. Can we create a ".NET terminology" topic to house that content? It pains me to include that content in an intro. We need to balance between servicing experienced and new .NET developers, and some of that content leans too heavily to addressing the needs of experiences .NET developers. For example, "CLI" vs "CLI" isn't something that comes up much (with any audience).

richlander committed:
Apply suggestions from code review

Co-authored-by: Jan Kotas <jkotas@microsoft.com>
Co-authored-by: Tom Dykstra <tdykstra@microsoft.com>
Co-authored-by: Beth Massi <bethma@microsoft.com>

richlander commented on docs/core/introduction.md:
```diff
@@ -8,167 +8,61 @@ recommendations: false
-  
-* [Visual Basic](../visual-basic/index.yml)
+* [Console apps](tutorials/with-visual-studio-code.md)
+* Windows Desktop apps
```
```suggestion
* [Windows Desktop apps](https://dotnet.microsoft.com/apps/desktop)
```

richlander committed:
Update docs/core/introduction.md

richlander commented on docs/core/introduction.md:
```diff
@@ -8,167 +8,61 @@ recommendations: false
-* [C#](../csharp/index.yml)
-
-  C# (pronounced "See Sharp") is a modern, object-oriented, and type-safe programming language. C# has its roots in the C family of languages and will be immediately familiar to C, C++, Java, and JavaScript programmers.
+* [Mobile apps](https://dotnet.microsoft.com/apps/maui)
```
```suggestion
* [Desktop apps](https://dotnet.microsoft.com/apps/desktop)
* [Mobile apps](https://dotnet.microsoft.com/apps/maui)
```

richlander committed:
Update docs/core/introduction.md

tdykstra commented:
> Can we create a ".NET terminology" topic to house that content? 

Much of the info is already in the glossary, perhaps all we need is a list of particularly problematic terms in the glossary introduction, with links to the relevant h2 sections in the glossary. This article could then just link to the glossary for info about .NET terminology.

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +74,162 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+The CLR also offers some "native code" capabilities, primarily oriented on performance. Applications can be ahead-of-time compiled to avoid most of the cost of the JIT at runtime. The CLR offers low-level C-style interop functionality, via a combination of [P/Invoke](../standard/native-interop/index.md), value types, and the ability to [blit](../framework/interop/blittable-and-non-blittable-types ...
 
-* Publishing an app as *self-contained* produces an executable file that includes the .NET [runtime](#sdk-and-runtimes) and [libraries](#runtime-libraries), and the application and its dependencies. Users of the application can run it on a machine that doesn't have the .NET runtime installed. Self-contained apps are platform-specific, and they can optionally be published using a form of [AOT comp ...
+The CLR enables access to native memory and pointer arithmetic via [`unsafe` code](../csharp/language-reference/unsafe-code.md). These operations are needed for certain algorithms, system interoperability, or to implement the most efficient algorithm. Unsafe code might not execute the same way in different environments and also loses the benefits of the garbage collector and type safety. We recom ...
```
```suggestion
The CLR enables access to native memory and pointer arithmetic via [`unsafe` code](../csharp/language-reference/unsafe-code.md). These operations are needed for certain algorithms, system interoperability, or to implement the most efficient algorithm. Unsafe code might not execute the same way in different environments and also loses the benefits of garbage collection and type safety. We recommend that you avoid unsafe code and centralize it as much as possible when it's used.
```

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +74,162 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+The CLR includes a Just-In-Time (JIT) compiler that compiles IL to native code at run time, targeting the underlying operating system and hardware. The JIT (called RyuJIT) can compile code at higher or lower levels of quality to enable better startup and steady-state throughput performance.
 
-.NET apps can be published in two different modes:
+The CLR also offers some "native code" capabilities, primarily oriented on performance. Applications can be ahead-of-time compiled to avoid most of the cost of the JIT at runtime. The CLR offers low-level C-style interop functionality, via a combination of [P/Invoke](../standard/native-interop/index.md), value types, and the ability to [blit](../framework/interop/blittable-and-non-blittable-types ...
```
Could say more about AOT, like the info in the [AOT compiler](https://docs.microsoft.com/en-us/dotnet/core/introduction#aot-compiler) section of the current version of the article.

richlander replied:
I think we should just delete the AOT compiler section. It is super confusing and deserves its own topic.

richlander replied:
I added a "Compilation" section that covers JIT and AOT topics and is accurate across the whole portfolio.

https://github.com/dotnet/docs/pull/29275/commits/9efeb4ae0146187a48736ba2b6f3fbbdea9f0120#diff-8a075dcee56c0a2dfdc0ede4f987a8ec54a370e3b34f468a253dffbe7dd26f67R156

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +74,162 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+[NuGet](/nuget/what-is-nuget) is the package manager for .NET. It enables developers to share compiled binaries with each other. [NuGet.org](https://www.nuget.org/) offers many [popular packages](https://www.nuget.org/stats/packages) from the community.
 
-Libraries for some commonly used application functionality aren't included in the runtime libraries but are made available in NuGet packages, such as the following:
+The following table lists NuGet packages from the .NET Team:
```
```suggestion
The following table lists NuGet packages from the .NET team:
```

tdykstra reviewed (commented):
A few more suggestions.

richlander commented on docs/core/introduction.md:
```diff
@@ -179,231 +74,162 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
 
-Higher-level .NET languages, such as C#, compile down to a hardware-agnostic instruction set, which is called Intermediate Language (IL). When an app runs, the JIT compiler translates IL to machine code that the processor understands. JIT compilation happens on the same machine that the code is going to run on.
+* [.NET SDK](sdk.md) -- enables development and CI scenarios.
+* [.NET Runtimes](https://dotnet.microsoft.com/en-us/download/dotnet/6.0/runtime) -- required for framework-dependent apps for test or production deployment. For more information, see [Deployment models](#deployment-models) later in this article.
```
This phrasing doesn't ring true for me. Why do we need to say "w/o development tools"? Also, it's missing the point that the runtime is included. Also, the `dotnet` tools *is* included with all Runtimes.

"Sets of libraries" doesn't make sense to me as a phrase. Libraries is already plural. Additional grouping (described by each "set") doesn't seem important.

```suggestion
* [.NET Runtimes](https://dotnet.microsoft.com/en-us/download/dotnet/6.0/runtime) -- .NET host, runtime and libraries, for test and production scenarios. You install a runtime on a machine to provide the execution environment for [framework-dependent apps](#deployment-models).
```

tdykstra replied:
"Sets" was an attempt to have the description be plural to be consistent with "Runtimes" plural, in the sense of .NET Runtime, Desktop Runtime, and ASP.NET Core Runtime. Including "runtime" in the definition (".NET host, runtime, and libraries"), in a different sense than "Runtime" at the head of the entry is likely to be confusing, and .NET host would need explanation at this point also. "w/o dev tools" was an attempt to call out the most important difference between SDK and Runtimes, at a level easily understood by someone not familiar with .NET. The dotnet driver included with the runtime isn't intended to be a tool used for development.

richlander replied:
https://github.com/dotnet/docs/pull/29275/commits/3230849247dcf0cfc77aa5b918dd98adc3cef6c3

richlander committed:
Apply suggestions from code review

Co-authored-by: Tom Dykstra <tdykstra@microsoft.com>

richlander commented on docs/core/introduction.md:
```diff
@@ -8,167 +8,62 @@ recommendations: false
+.NET is a free, cross-platform, open source developer platform for building applications. .NET is built on a [high-performance runtime](https://devblogs.microsoft.com/dotnet/category/performance/) that is used in production by many [high-scale apps](https://devblogs.microsoft.com/dotnet/category/developer-stories/).
+
+You can [download .NET](https://dotnet.microsoft.com/download) and use it for building many types of apps.
+
```
I don't think most users care about that as their first concern. Let's focus on the value the platform offers. I'm trying to push all the "fine print" details lower (or remove them entirely).

richlander committed:
Update AOT content

richlander commented on docs/core/introduction.md:
```diff
@@ -179,231 +74,162 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+You can [download .NET](https://dotnet.microsoft.com/download/dotnet/6.0) for development or production scenarios:
 
-Higher-level .NET languages, such as C#, compile down to a hardware-agnostic instruction set, which is called Intermediate Language (IL). When an app runs, the JIT compiler translates IL to machine code that the processor understands. JIT compilation happens on the same machine that the code is going to run on.
+* [.NET SDK](sdk.md) -- enables development and CI scenarios.
```
```suggestion
* [.NET SDK](sdk.md) -- Tools, libraries, and runtimes for development, testing, and CI scenarios.
```

richlander committed:
Apply suggestions from code review

Co-authored-by: Tom Dykstra <tdykstra@microsoft.com>

richlander committed:
Add MAR link

richlander committed:
Add iOS link

richlander committed:
Add R2R link

richlander committed:
Add diagnostics section

richlander committed:
Raised distributions section

richlander committed:
Add diagnostics links

richlander committed:
Update support statements

richlander committed:
Change runtime and sdk description

richlander commented on docs/core/introduction.md:
```diff
@@ -179,231 +77,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
-Libraries for some commonly used application functionality aren't included in the runtime libraries but are made available in NuGet packages, such as the following:
+[NuGet](/nuget/what-is-nuget) is the package manager for .NET. It enables developers to share compiled binaries with each other. [NuGet.org](https://www.nuget.org/) offers many [popular packages](https://www.nuget.org/stats/packages) from the community.
+
+The following table lists NuGet packages from the .NET team:
```
Good call. Are they the most important? This is the existing table. I didn't try to improve it. Should we just nuke it?

jkotas replied:
Either nuke it, or change it to be a table of most popular nuget packages (based on nuget.org download counts).

terrajobst replied:
Nuking seems fine; I don't think the ones listed are the most important nor the most representative. It would entirely depend on the kind of app you're building IMHO.

We could point to the .NET team NuGet user names but there are also a lot of packages that we don't really expect developers to reference (e.g. all the runtime and ref packs). Sadly, those are downloaded a lot hence they tend to rank highly and thus paint a very "deep in the weeds" or "gory implementation details".

*If* we want to include some NuGet packages my recommendation would be to pick packages that are app model neutral, generally referenced as a package (as opposed to being auto referenced as part of a shared framework), and not legacy (provided for backwards compatibility and/or porting from .NET Framework).

Good candidates would be:

* EF Core (`Microsoft.EntityFrameworkCore`)
* SQL Client (`Microsoft.Data.SqlClient`)
* IoT (`System.Device.Gpio`)

richlander replied:
I nuked it. We can come with something different if we want later.

jkotas commented on docs/core/introduction.md:
```diff
@@ -8,167 +8,65 @@ recommendations: false
-* [Delegates](../standard/delegates-lambdas.md)
-* [Lambdas](../standard/delegates-lambdas.md)
+* [Asynchronous code](../csharp/programming-guide/concepts/async/index.md)
+* [Attributes](../standard/attributes/index.md)
```
This list should include Reflection

richlander replied:
```suggestion
* [Attributes](../standard/attributes/index.md)
* Reflection
```

It appears that we only have a .NET Framework version of this doc. Logged an issue https://github.com/dotnet/docs/issues/29741

jeffschwMSFT commented on docs/core/introduction.md:
```diff
@@ -179,231 +77,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+## Binary distributions
 
-.NET Interactive is a group of CLI tools and APIs that enable users to create interactive experiences across the web, markdown, and notebooks.
+You can [download .NET](https://dotnet.microsoft.com/download/dotnet/6.0) for development or production scenarios:
```
I would not use a specific version in the link, unless we are signing up for an annual reminder to update to the latest 'current'.

jeffschwMSFT replied:
There are a few 6.0 links that my commend equally applies too.

richlander replied:
```suggestion
You can [download .NET](https://dotnet.microsoft.com/download/dotnet) for development or production scenarios:
```

jeffschwMSFT commented on docs/core/introduction.md:
```diff
@@ -179,231 +77,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
 
-* Some scenarios require 100% AOT compilation. An example is [iOS](/xamarin/ios/).
-* In other scenarios, most of an app's code is AOT-compiled but some is JIT-compiled. Some code patterns aren't friendly to AOT (like generics). An example of this form of AOT compilation is the [ready-to-run](whats-new/dotnet-core-3-0.md#readytorun-images) publish option. This form of AOT offers the benefits of AOT without its drawbacks.
+## Runtime technology
```
Not sure if it is critical for this intro doc, but there are runtime technology(s) in our stack: coreclr, mono, native aot, etc.  Perhaps a note that the runtime used is tailored to the scenario.

richlander replied:
There was more of that in the current doc, but it is super confusing. I think a new doc with that info is warranted.

https://docs.microsoft.com/en-us/dotnet/core/introduction#net-core-net-framework-mono-uwp

richlander replied:
I added a (starter) section at the end.

richlander commented on docs/core/introduction.md:
```diff
@@ -8,167 +8,65 @@ recommendations: false
-* [Runtime libraries](#runtime-libraries). Provides primitive data types and fundamental utilities.
-* The `dotnet` [driver](tools/index.md#driver). A CLI command that runs framework-dependent apps.
+* [Code analyzers](../fundamentals/code-analysis/overview.md)
+* [Source Link (source maps)](https://github.com/dotnet/sourcelink/blob/main/README.md)
```
```suggestion
```

richlander committed:
Update docs/core/introduction.md

Co-authored-by: Jan Kotas <jkotas@microsoft.com>

danmoseley commented on docs/core/introduction.md:
```diff
@@ -8,167 +8,64 @@ recommendations: false
+## Features
 
-Here are some of the capabilities that .NET languages support:
+.NET includes many features that enable developers to productively write reliable and performant code.
```
```suggestion
.NET features allow developers to productively write reliable and performant code.
```
less wishy washy?

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+.NET source and and binaries are licensed with the [MIT license](https://github.com/dotnet/runtime/blob/main/LICENSE.TXT). Additional [licenses apply on Windows](https://github.com/dotnet/core/blob/main/license-information-windows.md) for binary distributions.
 
-For more information, see [Common Language Runtime (CLR) overview](../standard/clr.md).
+You can get the .NET SDK and runtimes from the following sources:
```
source and binaries? little unclear what the difference is between these and the binaries links above.

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+## Support
 
-Since JIT compilation occurs during execution of the application, the compilation time is part of the run time. Therefore, JIT compilers have to balance time spent optimizing code against the savings that the resulting code can produce. But a JIT compiler knows the actual hardware and can free developers from having to ship different implementations for different platforms.
+[Microsoft supports .NET](https://github.com/dotnet/core/blob/main/os-lifecycle-policy.md) on Android, Apple, Linux, and Windows operating systems. It can be used on Arm64, x64, and x86 chips. It is also supported in emulated environments, like [macOS Rosetta 2](https://support.apple.com/HT211861).
```
do we still say 'chips' ? I rarely see it written these days. Maybe 'systems' or 'architectures'

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+[Microsoft supports .NET](https://github.com/dotnet/core/blob/main/os-lifecycle-policy.md) on Android, Apple, Linux, and Windows operating systems. It can be used on Arm64, x64, and x86 chips. It is also supported in emulated environments, like [macOS Rosetta 2](https://support.apple.com/HT211861).
 
-The .NET JIT compiler can do *tiered compilation*, which means it can recompile individual methods at run time. This feature lets it compile quickly while still being able to produce a highly tuned version of the code for frequently used methods.
+New versions of .NET are released annually in November. .NET releases in odd-numbered years are Long-term Support (LTS) releases and are supported for three years. Releases in even-numbered years are Short-team Support (STS) releases and are supported for 18 months. All other aspects of the releases are the same, like quality level and breaking change policies. For more information, see [Releases ...
```
```suggestion
New versions of .NET are released annually in November. .NET releases in odd-numbered years are Long-term Support (LTS) releases and are supported for three years. Releases in even-numbered years are Short-team Support (STS) releases and are supported for 18 months. The quality level, breaking change policies, and all other aspects of the releases are the same. For more information, see [Releases and support](releases-and-support.md).
```

danmoseley replied:
leading with the lede

richlander commented on docs/core/introduction.md:
```diff
@@ -8,167 +8,64 @@ recommendations: false
 # What is .NET? Introduction and overview
 
-.NET is a free, open-source development platform for building many kinds of apps, such as:
+.NET is a free, cross-platform, open source developer platform for building applications. .NET is built on a [high-performance runtime](https://devblogs.microsoft.com/dotnet/category/performance/) that is used in production by many [high-scale apps](https://devblogs.microsoft.com/dotnet/category/developer-stories/).
```
```suggestion
.NET is a free, cross-platform, open source developer platform for building many kinds of applications. .NET is built on a [high-performance runtime](https://devblogs.microsoft.com/dotnet/category/performance/) that is used in production by many [high-scale apps](https://devblogs.microsoft.com/dotnet/category/developer-stories/).
```

danmoseley replied:
LGTM

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+## Runtime technology
 
-### Automatic memory management
+The [Common Language Runtime (CLR)](../standard/clr.md) provides capabilities and services to apps that govern their function and also define their basic security and reliability behavior. The [fundamental features of the runtime](https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/intro-to-clr.md) are:
```
```suggestion
The [Common Language Runtime (CLR)](../standard/clr.md) is the foundation all .NET apps are built on. The [fundamental features of the runtime](https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/intro-to-clr.md) are:
```
although, there's Mono. should it just say "runtime" and describe CoreCLR/mono as instances separately?

richlander commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+[Red Hat supports .NET](https://developers.redhat.com/topics/dotnet/) on Red Hat Enterprise Linux (RHEL).
 
-The default experience for most .NET workloads is the JIT compiler, but .NET offers two forms of ahead-of-time (AOT) compilation:
+[Samsung supports .NET](https://developer.tizen.org/development/training/.net-application) on Tizen platforms.
```
We could. I was thinking of corporate-supported but if there is an *active* community group, then sure.

danmoseley replied:
@wfurt how would you characterize the level of community support for FreeBSD?

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+* Cross-platform design
 
-The GC is one of the CLR services that help ensure *memory safety*. A program is memory safe if it accesses only allocated memory. For instance, the runtime ensures that an app doesn't access unallocated memory beyond the bounds of an array.
+.NET is sometimes called a "managed code" runtime. It is called *managed* primarily because it uses a garbage collector for memory management and because it enforces type and memory safety. The CLR is also called a *virtual machine*. It virtualizes (or abstracts) a variety of operating system and hardware concepts, such as memory, threads, and exceptions.
```
I don't think the cognitive load of 'virtual machine' is worth introducing here. It has a very clear and different meaning in most people's minds. Back when .NET was created, we called it the VM but virtual machines in today's sense were (IIRC) not something most developers were exposed to. Even on our team, some old timers call it the VM but most do not call it that and I have never heard anyone on the team refer to it as a virtual machine.

richlander replied:
The JVM is the counter example. That said, I can remove this.

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+.NET is sometimes called a "managed code" runtime. It is called *managed* primarily because it uses a garbage collector for memory management and because it enforces type and memory safety. The CLR is also called a *virtual machine*. It virtualizes (or abstracts) a variety of operating system and hardware concepts, such as memory, threads, and exceptions.
 
-For more information, see [Automatic memory management](../standard/automatic-memory-management.md) and [Fundamentals of garbage collection](../standard/garbage-collection/fundamentals.md).
+The CLR was designed to be a cross-platform runtime from its inception. It has been ported to multiple operating systems and chip architectures. Cross-platform .NET code typically does not need to be recompiled to run in new environments. Instead, you just need to install a different runtime to run your app.
```
same comment on chip. I rarely hear that used, mostly people seem to say architecture or machine.

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
-.NET apps can be published in two different modes:
+* [F#](../fsharp/index.yml) is an interoperable programming language for writing succinct, robust, and performant code. F# programming is data-oriented, where code involves transforming data with functions.
+  
+* [Visual Basic](../visual-basic/index.yml) uses a more verbose syntax that is closer to ordinary human language. It can be an easier language to learn for people new to programming.
```
Is that true, or just a hypothesis? I have not seen evidence of this (although, I could well simply have not seen it)

The principal reason to use VB I would think is existing familiarity with VB or other Basic-style languages.

richlander replied:
This was the text from before. Thoughts @KathleenDollard?

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+.NET apps (as written in a high-level language like C#) are compiled to an [Intermediate Language (IL)](https://en.wikipedia.org/wiki/Common_Intermediate_Language). IL is a compact code format that can be supported on any operating system or chip hardware. Most .NET apps use APIs that are supported in multiple environments, requiring only the .NET runtime to run.
 
-  You can install multiple versions of the runtime side by side to run framework-dependent apps that target different versions of the runtime. For more information, see [Target frameworks](../standard/frameworks.md).
+IL needs to be compiled to native code in order to execute on a CPU, for example Arm64 or x64. .NET supports both Ahead-Of-Time (AOT) and Just-In-Time (JIT) compilation models. For example, on Android, Linux, macOS, and Linux, JIT compilation is the default, and AOT is optional (for example, with [ReadyToRun](deploy/ready-to-run.md)). On [iOS](/xamarin/ios/), AOT is mandatory (except when running ...
```
this almost seems like a table would be a better presentation.

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+IL needs to be compiled to native code in order to execute on a CPU, for example Arm64 or x64. .NET supports both Ahead-Of-Time (AOT) and Just-In-Time (JIT) compilation models. For example, on Android, Linux, macOS, and Linux, JIT compilation is the default, and AOT is optional (for example, with [ReadyToRun](deploy/ready-to-run.md)). On [iOS](/xamarin/ios/), AOT is mandatory (except when running ...
 
-Executables are produced for specific target platforms, which you specify with a [runtime identifier (RID)](rid-catalog.md).
+The advantage of the JIT is that it can compile an app (unmodified) to the CPU instructions and calling conventions in a given environment, per the underlying operating system and hardware. It can also compile code at higher or lower levels of quality to enable better startup and steady-state throughput performance.
```
```suggestion
The advantage of JIT compilation is that it can generate code for the specific hardware that the program is running on. It also has the opportunity to dynamically vary the level of optimization of the code it generates based on observations as the program executes - for example to favor faster startup over faster steady-state throughput performance.
```

richlander replied:
Not quite. Tiering doesn't have to favor startup over faster steady state. It gives both.

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
 ## Runtime libraries
 
-.NET has an expansive standard set of class libraries, known as [runtime libraries](../standard/glossary.md#runtime), [framework libraries](../standard/glossary.md#framework-libraries), or the [base class library (BCL)](../standard/glossary.md#bcl). These libraries provide implementations for many general-purpose and workload-specific types and utility functionality.
+.NET has an expansive standard set of class libraries. These libraries provide implementations for many general-purpose and workload-specific types and utility functionality.
```
```suggestion
.NET has a comprehensive standard set of class libraries. These libraries provide implementations for many general-purpose and workload-specific types and utility functionality.
```
or extensive

danmoseley commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+In 2002, Microsoft released [.NET Framework](../framework/get-started/overview.md), a development platform for creating Windows apps. Today .NET Framework is at version 4.8 and remains [supported by Microsoft](https://dotnet.microsoft.com/platform/support/policy/dotnet-framework).
 
-For more information, see [Unsafe code and pointers](../csharp/language-reference/unsafe-code.md).
+In 2014, Microsoft introduced .NET Core as a cross-platform, open-source successor to .NET Framework. This new [implementation of .NET](../standard/glossary.md#implementation-of-net) kept the name .NET Core through version 3.1. The next version after .NET Core 3.1 was named .NET 5. New .NET versions continue to be released annually, with a growing set of features and supported scenarios.
```
maybe worth a reminder that the major version # increments by 1 each year

richlander commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+In 2002, Microsoft released [.NET Framework](../framework/get-started/overview.md), a development platform for creating Windows apps. Today .NET Framework is at version 4.8 and remains [supported by Microsoft](https://dotnet.microsoft.com/platform/support/policy/dotnet-framework).
 
-For more information, see [Unsafe code and pointers](../csharp/language-reference/unsafe-code.md).
+In 2014, Microsoft introduced .NET Core as a cross-platform, open-source successor to .NET Framework. This new [implementation of .NET](../standard/glossary.md#implementation-of-net) kept the name .NET Core through version 3.1. The next version after .NET Core 3.1 was named .NET 5. New .NET versions continue to be released annually, with a growing set of features and supported scenarios.
```
@terrajobst tells us 2014 -> https://devblogs.microsoft.com/dotnet/net-core-is-open-source/

richlander commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+The advantage of the JIT is that it can compile an app (unmodified) to the CPU instructions and calling conventions in a given environment, per the underlying operating system and hardware. It can also compile code at higher or lower levels of quality to enable better startup and steady-state throughput performance.
 
-For more information, see [.NET application publishing overview](deploying/index.md) and [Introduction to .NET and Docker](docker/introduction.md).
+The advantage of AOT is that it provides the best app startup and can result in smaller deployments. The primary downside is that binaries must be built for each separate deployment target (the same as any other native code).
```
This is where I'm probably conflating things. What I meant (and didn't say) is that for CoreCLR apps, the trimmer can only trim so far. For NativeAOT, it can trim much farther (into what we think of as the runtime).

richlander replied:
Unless you are referring to NGEN, our AOT story doesn't really have versioning problems. Right?

richlander committed:
Apply suggestions from code review

Co-authored-by: Dan Moseley <danmose@microsoft.com>
Co-authored-by: Jan Kotas <jkotas@microsoft.com>
Co-authored-by: Jeff Schwartz <jeffschw@microsoft.com>

richlander commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+The [.NET SDK](sdk.md) is a set of libraries and [tools](tools/index.md) for developing and running .NET applications. It includes the [MSBuild](/visualstudio/msbuild/msbuild) build engine, the [Roslyn](https://github.com/dotnet/roslyn) (C# and Visual Basic) compiler, and the [F#](https://github.com/microsoft/visualfsharp) compiler. Most commands are run by using the [`dotnet` command](tools/dotn ...
 
-* **.NET**
+The [Visual Studio](https://visualstudio.microsoft.com/) family of IDEs offer excellent support for .NET and the C#, F#, and Visual Basic languages.
```
Good point. I defined it (just now) earlier in the document.

richlander committed:
Update per feedback

richlander commented on docs/core/introduction.md:
```diff
@@ -179,231 +76,172 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+## Languages
 
-For more information, see [Cleaning up unmanaged resources](../standard/garbage-collection/unmanaged.md).
+The CLR is designed to support multiple programming languages. C#, F#, and Visual Basic languages are supported by Microsoft and are designed in collaboration with the community.
```
I added a section in the end about that.

richlander committed:
Add .NET ecosystem section

richlander committed:
Add links to intro

richlander committed:
Merge branch 'main' into intro

richlander committed:
Fix links

richlander committed:
Fix links

richlander commented:
Hey @tdykstra ... I fixed the remaining errors. Is it ready to merge?

richlander commented on docs/core/introduction.md:
```diff
@@ -179,231 +72,175 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+[Samsung supports .NET](https://developer.tizen.org/development/training/.net-application) on Tizen platforms.
 
-The default experience for most .NET workloads is the JIT compiler, but .NET offers two forms of ahead-of-time (AOT) compilation:
+## Runtime technology
```
```suggestion
## Runtime
```

richlander committed:
Update docs/core/introduction.md

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +72,175 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+[Microsoft supports .NET](https://github.com/dotnet/core/blob/main/os-lifecycle-policy.md) on Android, Apple, Linux, and Windows operating systems. It can be used on Arm64, x64, and x86 architectures. It is also supported in emulated environments, like [macOS Rosetta 2](https://support.apple.com/HT211861).
 
-Since JIT compilation occurs during execution of the application, the compilation time is part of the run time. Therefore, JIT compilers have to balance time spent optimizing code against the savings that the resulting code can produce. But a JIT compiler knows the actual hardware and can free developers from having to ship different implementations for different platforms.
+New versions of .NET are released annually in November. .NET releases in odd-numbered years are Long-term Support (LTS) releases and are supported for three years. Releases in even-numbered years are Short-team Support (STS) releases and are supported for 18 months. The quality level, breaking change policies, and all other aspects of the releases are the same. For more information, see [Releases ...
```
```suggestion
New versions of .NET are released annually in November. .NET releases in odd-numbered years are Long-term Support (LTS) releases and are supported for three years. Releases in even-numbered years are Short-term Support (STS) releases and are supported for 18 months. The quality level, breaking change policies, and all other aspects of the releases are the same. For more information, see [Releases and support](releases-and-support.md).
```

tdykstra committed:
Update docs/core/introduction.md

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +72,175 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+You can download .NET from:
 
-.NET apps run [managed code](../standard/managed-code.md) in a runtime environment known as the Common Language Runtime (CLR).
+* [The Microsoft download site](https://dotnet.microsoft.com/download/)
+* [Containers](https://mcr.microsoft.com/catalog?search=dotnet/)
+* [Linux package managers](install/linux.md).
```
```suggestion
* [The Microsoft download site](https://dotnet.microsoft.com/download/).
* [Containers](https://mcr.microsoft.com/catalog?search=dotnet/).
* [Linux package managers](install/linux.md).
```

tdykstra committed:
Update docs/core/introduction.md

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +72,175 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+The [Common Language Runtime (CLR)](../standard/clr.md) is the foundation all .NET apps are built on. The [fundamental features of the runtime](https://github.com/dotnet/runtime/blob/main/docs/design/coreclr/botr/intro-to-clr.md) are:
 
-### Automatic memory management
+* Garbage collection
+* Memory safety and type safety
+* High level support for programming languages
+* Cross-platform design
```
```suggestion
* Garbage collection.
* Memory safety and type safety.
* High level support for programming languages.
* Cross-platform design.
```

tdykstra committed:
Update docs/core/introduction.md

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +72,175 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
+* Cross-platform design.
 
-The *garbage collector* (GC) manages the allocation and release of memory for applications. Each time your code creates a new object, the CLR allocates memory for the object from the [managed heap](../standard/garbage-collection/fundamentals.md#the-managed-heap). As long as address space is available in the managed heap, the runtime continues to allocate space for new objects. When not enough fre ...
+.NET is sometimes called a "managed code" runtime. It is called *managed* primarily because it uses a garbage collector for memory management and because it enforces type and memory safety. The CLR is also virtualizes (or abstracts) a variety of operating system and hardware concepts, such as memory, threads, and exceptions.
```
```suggestion
.NET is sometimes called a "managed code" runtime. It is called *managed* primarily because it uses a garbage collector for memory management and because it enforces type and memory safety. The CLR virtualizes (or abstracts) a variety of operating system and hardware concepts, such as memory, threads, and exceptions.
```

tdykstra committed:
Update docs/core/introduction.md

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +72,175 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
-  You can install multiple versions of the runtime side by side to run framework-dependent apps that target different versions of the runtime. For more information, see [Target frameworks](../standard/frameworks.md).
+* On Android, Linux, macOS, and Linux, JIT compilation is the default, and AOT is optional (for example, with [ReadyToRun](deploying/ready-to-run.md)).
+* On [iOS](/xamarin/ios/), AOT is mandatory (except when running in the simulator).
+* In Wasm environments, AOT is mandatory.
```
```suggestion
* In WebAssembly (Wasm) environments, AOT is mandatory.
```

tdykstra committed:
Update docs/core/introduction.md

tdykstra commented on docs/core/introduction.md:
```diff
@@ -179,231 +72,175 @@ A .NET app is built from source code by using [MSBuild](/visualstudio/msbuild/ms
-For more information, see [Unsafe code and pointers](../csharp/language-reference/unsafe-code.md).
+* **.NET Framework** -- It provides access to the broad capabilities of Windows and Windows Server. Also extensively used for Windows-based cloud computing. The original .NET.
+* **Mono** -- A cross-platform implementation of .NET Framework. The original community and open source .NET. Used for Android, iOS, and Wasm apps.
+* **.NET (Core)** -- A cross-platform and open source implementation of .NET, re-thought for the cloud age while remaining significantly compatible with .NET Framework. Uses for Linux, macOS, and Windows support.
```
```suggestion
* **.NET (Core)** -- A cross-platform and open source implementation of .NET, rethought for the cloud age while remaining significantly compatible with .NET Framework. Used for Linux, macOS, and Windows apps.
```

tdykstra committed:
Update docs/core/introduction.md

tdykstra commented:
There were 5 unresolved conversations -- I marked them as resolved so we can merge. I also did a final edit pass and made some final mostly style corrections. Made a slight wording change in the new ecosystem section to fix text that didn't make sense as it was. You can open a new PR if any of my last-minute updates need work. Thanks for all your help with this article!

richlander commented:
Sounds good @tdykstra. Really appreciate all the help landing this one. Have to figure out what to look at next.
